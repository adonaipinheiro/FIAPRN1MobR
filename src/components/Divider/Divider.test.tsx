import React from 'react';
import renderer from 'react-test-renderer';

import {Divider} from './';

const props = {
  size: 16,
};

const propsHorizontal = {
  size: 16,
  horizontal: true,
};

describe('<Divider />', () => {
  it('Should render corretly', () => {
    const divider = renderer.create(<Divider {...props} />);
    expect(divider).toBeTruthy();
  });

  it('Should render corretly with horizontal', () => {
    const divider = renderer.create(<Divider {...propsHorizontal} />);
    expect(divider).toBeTruthy();
  });

  it('Should render corretly with size', () => {
    const inst = renderer.create(<Divider {...propsHorizontal} />);
    const size = inst.root.props.size;
    expect(size).toBe(16);
  });

  it('Should take Snapshot', () => {
    const snapshot = renderer.create(<Divider {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
