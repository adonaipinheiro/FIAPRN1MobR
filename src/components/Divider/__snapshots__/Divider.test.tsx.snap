// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> Should take Snapshot 1`] = `
<View
  style={
    {
      "height": 16,
      "width": 0,
    }
  }
/>
`;
